zebedee-cms/src/test/java/com/github/onsdigital/zebedee/model/publishing/PublishNotificationTest.java

```diff
@@ -0,0 +1,387 @@
+
+        boolean hasExpectedPaths = payloadCapture.getAllValues().stream().map(payload -> payload.uriToUpdate).allMatch(expectedPaths::contains);
+
+        assertTrue(hasExpectedPaths);
+
+        boolean allDatesAreNull = payloadCapture.getAllValues().stream().map(payload -> payload.publishDate).allMatch(Objects::isNull);
+
+        assertTrue(allDatesAreNull);
+    }
+
+    @Test
+    public void publishNotificationWhenCacheAPIEnabledAndPUBLISHTest() throws IOException {
+        when(Configuration.isLegacyCacheAPIEnabled()).thenReturn(true);
+
+        assertEquals(3, collectionWithDeletes.reviewedUris().size());
+        assertEquals(2, collectionWithDeletes.getDescription().getPendingDeletes().size());
+
+        PublishNotification publishNotification = new PublishNotification(collectionWithDeletes);
+        assertEquals(5, publishNotification.getLegacyCacheApiPayloads().size());
+
+        Http mockHttp = mock(Http.class);
+        ProtocolVersion version = new ProtocolVersion("1", 1, 0);
+        BasicStatusLine basicStatusLine = new BasicStatusLine(version, 200, "OK");
+
+        WebsiteResponse websiteResponse = new WebsiteResponse();
+        websiteResponse.setMessage("OK");
+        Response<Object> response = new Response<>(basicStatusLine, websiteResponse);
+        when(mockHttp.put(any(), any(), any(), any())).thenReturn(response);
+
+        PayloadsByEvent payloadsByEvent = new PayloadsByEvent("cake-1-1234", EventType.PUBLISHED, publishNotification.getLegacyCacheApiPayloads());
+
+        LegacyCacheApiClient.sendPayloads(mockHttp, hosts.get(0), payloadsByEvent);
+
+        verify(mockHttp, times(5)).put(endpointCapture.capture(), payloadCapture.capture(), classCapture.capture(), any());
+
+        List<String> expectedPaths = new ArrayList<>();
+        expectedPaths.add("/economy/inflationandprices/articles/producerpriceinflation/latest");
+        expectedPaths.add("/economy/inflationandprices/articles/producerpriceinflation/october2022");
+        expectedPaths.add("/economy/inflationandprices/articles/producerpriceinflation/october2023");
+        expectedPaths.add("/economy/inflationandprices/bulletins/producerpriceinflation/october2022");
+        expectedPaths.add("/economy/inflationandprices/bulletins/producerpriceinflation/october2023");
+
+        boolean hasExpectedPaths = payloadCapture.getAllValues().stream().map(payload -> payload.uriToUpdate).allMatch(expectedPaths::contains);
+
+        assertTrue(hasExpectedPaths);
+
+        Long expectedNumberOfPayloadsWithoutPublishDate = 2L;
+        Long payloadsWithoutPublishDate = payloadCapture.getAllValues().stream().filter(payload -> payload.publishDate == null).count();
+
+        assertEquals(expectedNumberOfPayloadsWithoutPublishDate, payloadsWithoutPublishDate);
+    }
+
+    @Test
+    public void publishNotificationNotClientCallWhenNoUrlsWhenCacheAPIEnabledTest() throws IOException {
```

Comment:
```suggestion
    public void publishNotificationNoClientCallsWhenNoUrlsWhenCacheAPIEnabledTest() throws IOException {
```

Is that more descriptive?